### PR TITLE
Yet more compiler improvements and fixes

### PR DESCRIPTION
**Pull request by ben-clayton**

### CODE REVIEW

File: core/codegen/debug.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+func (d *dbg) builder() *llvm.DIBuilder {
+	if d.llvm != nil {
+		return d.llvm
+	}
+
+	b := llvm.NewDIBuilder(d.m.llvm)
+	d.cu = b.CreateCompileUnit(llvm.DICompileUnit{
+		Language:       0x0001, // DW_LANG_C,
+		File:           "apis",
+		Dir:            "/ssd/src/gapid", // TODO: HACK
```
Comment (AWoloszyn) [inline core/codegen/debug.go]: do we want this here?

File: core/codegen/module.go [hunk 2]
```diff
@@ -98,6 +103,13 @@ func NewModule(name string, target *device.ABI) *Module {
 	m.Types.emptyStructField = Field{"empty-struct", m.Types.Int8}
 	m.Types.m = m
 
+	if true {
```
Comment (AWoloszyn) [inline core/codegen/module.go]: if true { ? 

Can we get a comment as to why?

Comment (ben-clayton) [reply core/codegen/module.go]: Good spot - bad merge!

File: core/codegen/debug.go (hunk 1 above)
Comment (ben-clayton) [reply core/codegen/debug.go]: Debug generation shouldn't be enabled (except you spotted it is below). I haven't figured out a nice way to get the project root here due to bazel complications. I'd like to revisit this once I have performance in the bag.

File: core/codegen/module.go (hunk 2 above)
Comment (ben-clayton) [reply core/codegen/module.go]: Fixed.

**Comment by ben-clayton** [comment]
Switch away from my LLVM fork, now that https://github.com/llvm-mirror/llvm/commit/e562960fe303c0ffab6f3458fcdb1544b56fd81e has been merged.

**Pull request merged**